app/models/user.rb

```diff
@@ -596,13 +596,15 @@ def create_personal_herbarium
   # Return an Array of SpeciesList's that User owns or that are attached to a
   # Project that the User is a member of.
   def all_editable_species_lists(include: nil)
-    @all_editable_species_lists ||= begin
-      return species_lists.includes(include) if projects_member.none?
+    @all_editable_species_lists ||= calc_all_editable_species_lists(include)
+  end
 
-      SpeciesList.includes(include).
-        where(SpeciesList[:user_id].eq(id).
-        or(SpeciesList[:id].in(species_lists_in_users_projects))).uniq
-    end
+  def calc_all_editable_species_lists(include: nil)
+    return species_lists.includes(include) if projects_member.none?
+
+    SpeciesList.includes(include).
+      where(SpeciesList[:user_id].eq(id).
+      or(SpeciesList[:id].in(species_lists_in_users_projects))).uniq
   end
```

Comment:
I think you can dispense with the ": nil" except in the first case...

3rd case: Since calc_all_editable_species_lists is a private method called only by the one method right above, you know that the include argument is always going to be passed, so no need to specify a default.  So you could change that to "def calc_all_editable_species_lists(include)".

2nd case: I think this is going to always pass in nil.  You want to pass include.  I guess it depends on how you handle the 3rd case.  The way you have it now, it uses named arguments, so it would be (include: include), but if you change it to just take a single mandatory argument, it would be shortened to just (include).

Does that make sense?

Reply:
PS. Thank you for agreeing to make this tiny change!

Reply:
Makes sense, and thanks for the explanation. Will merge if it passes tests.